src/entity.cpp

```diff
@@ -2576,78 +2578,108 @@ void Entity::handleEffects(Stat* myStats)
 		spawnAmbientParticles(20, 175, 20 + rand() % 30, 0.5, true);
 	}
 
-	// burning
+	// Process Burning Status Effect
 	if ( this->flags[BURNING] )
 	{
-		if ( ticks % 30 == 0 )
+		this->char_fire--; // Decrease the fire counter
+		messagePlayer(clientnum, "Time: %d, Chance: %d", this->char_fire, this->chanceToPutOutFire);
+		// Check to see if time has run out
+		if ( this->char_fire <= 0 )
 		{
-			this->modHP(-2 - rand() % 3);
-			if ( myStats->HP <= 0 )
+			this->flags[BURNING] = false;
+			messagePlayer(player, language[647]); // "The flames go out."
+			serverUpdateEntityFlag(this, BURNING);
+		}
+		else
+		{
+			// If 1 second has passed (50 ticks), process the Burning Status Effect
+			if ( (this->char_fire % TICKS_PER_SECOND) == 0 )
```

Comment:
50 ticks feels too slow now that I tested it, I think 30 was a good number after all.